lib/improver/utilities/pad_spatial.py

```diff
@@ -125,6 +128,44 @@ def create_cube_with_halo(cube, halo_radius):
     return halo_cube
 
 
+def remove_cube_halo(cube, halo_radius):
+    """
+    Remove halo of halo_radius from a cube.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube on extended grid
+        halo_radius (float):
+            Size of border to remove, in metres
+
+    Returns:
+        result (iris.cube.Cube):
+            New cube with the halo removed.
+    """
+    halo_size_x, halo_size_y = convert_distance_into_number_of_grid_cells(
+        cube, halo_radius)
+
+    original_attributes = cube.attributes
+    original_methods = cube.cell_methods
+    result_slices = iris.cube.CubeList()
+    for cube_slice in cube.slices([cube.coord(axis='y'),
+                                   cube.coord(axis='x')]):
+        cube_halo = remove_halo_from_cube(cube_slice,
+                                          halo_size_x,
+                                          halo_size_y)
+        result_slices.append(cube_halo)
+    result = result_slices.merge_cube()
+
+    result.cell_methods = original_methods
+    result.attributes = original_attributes
+    req_coords = []
+    for coord in cube.coords(dim_coords=True):
+        req_coords.append(coord.name())
+    result = enforce_coordinate_ordering(
+        result, req_coords, promote_scalar=True)
```

Comment:
Are we expecting the original coordinate order to be lost?

Reply:
This is being used to promote any scalar coord created in the merge back to a dimension coord. If there is a better way of doing this let me know.